operator/controllers/kyma_controller.go

```diff
@@ -208,24 +205,15 @@ func (r *KymaReconciler) HandleProcessingState(ctx context.Context, kyma *v1alph
 			fmt.Errorf("ParsedModule CR creation/update error: %w", err))
 	}
 
-	// Now we track the conditions: update the status based on their state
-	// technically we could also update the state in the previous step alone determine if we are ready based on this
-
 	statusUpdateRequiredFromModuleInfoSync := sync.SyncModuleInfo(ctx, kyma, modules)
-	kyma.SyncConditionsWithModuleStates()
-	if err != nil {
-		return r.UpdateStatusFromErr(ctx, kyma, v1alpha1.StateError,
-			fmt.Errorf("error while syncing conditions during processing: %w", err))
-	}
 
-	statusUpdateRequiredFromDeletion := r.UpdateModuleInfos(ctx, kyma)
 	moduleInfos := kyma.GetNoLongerExistingModuleInfos()
-	err = r.DeleteNoLongerExistingModules(ctx, moduleInfos)
-	if err != nil {
+	if err := r.DeleteNoLongerExistingModules(ctx, moduleInfos); err != nil {
```

Comment:
I think this needs a comment in the control loop,otherwise people wont understand why we need to check the delete case here

Reply:
added a comment